addons/crm_iap_lead_enrich/models/crm_lead.py

```diff
@@ -29,19 +30,21 @@ def _compute_show_enrich_button(self):
             else:
                 lead.show_enrich_button = True
 
-    @api.model
-    def _iap_enrich_leads_cron(self):
-        timeDelta = fields.datetime.now() - datetime.timedelta(hours=1)
-        # Get all leads not lost nor won (lost: active = False)
-        leads = self.search([
-            ('iap_enrich_done', '=', False),
-            ('reveal_id', '=', False),
-            ('probability', '<', 100),
-            ('create_date', '>', timeDelta)
-        ])
-        leads.iap_enrich(from_cron=True)
-
-    def iap_enrich(self, from_cron=False):
+    @api.model_create_multi
+    def create(self, vals_list):
+        leads = super(Lead, self).create(vals_list)
+        enrich_mode = self.env['ir.config_parameter'].get_param('crm.iap.lead.enrich.setting', 'manual')
+        if enrich_mode == 'auto' and (not self._context.get('install_mode') or self._context.get('import_file')) and not getattr(threading.currentThread(), 'testing', False):
+            leads_to_enrich = leads.filtered(lambda lead: lead.email_from)
+            if self._context.get('import_file'):
```

Comment:
Actually we could always apply the limit. No need to limit it with the import_file context key.